pkg/api/v1/persistentvolume/util.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package persistentvolume
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+func getClaimRefNamespace(pv *corev1.PersistentVolume) string {
+	if pv.Spec.ClaimRef != nil {
+		return pv.Spec.ClaimRef.Namespace
+	}
+	return ""
+}
+
+// Visitor is called with each object's namespace and name, and returns true if visiting should continue
+type Visitor func(namespace, name string, kubeletVisible bool) (shouldContinue bool)
+
+// VisitPVSecretNames invokes the visitor function with the name of every secret
+// referenced by the PV spec. If visitor returns false, visiting is short-circuited.
+// Returns true if visiting completed, false if visiting was short-circuited.
+func VisitPVSecretNames(pv *corev1.PersistentVolume, visitor Visitor) bool {
```

Comment:
Shouldn't this be paired with a delete of the existing internal one?

Reply:
if that's the only caller, yes

Reply:
oh good point, cleaning

Reply:
oops. internal ones are referenced by its REST 😅

[https://github.com/kubernetes/kubernetes/blob/91f93b74b17764fbf96bad1c26e8b53344169435/pkg/registry/core/persistentvolume/strategy.go#L56](https://github.com/kubernetes/kubernetes/blob/91f93b74b17764fbf96bad1c26e8b53344169435/pkg/registry/core/persistentvolume/strategy.go#L56)

Reply:
can delete VisitPVSecretNames and associated tests from internal, right?

Reply:
oh.. i thought we were talking about the entire pv util pkg..

Reply:
done